### DISCUSSION TITLE
writing itp after modification

### DISCUSSION THREAD

**Discussion by csbrasnett** [marrink-lab/vermouth-martinize#478] [General]
I'm reading in an itp file to modify it slightly, but I'm having trouble writing it out after doing so.

version is vermouth==0.7.3

here's a minimal example of what I'm trying to do:

```
ff = ForceField('path/to/martini3001')

file = 'in.itp'

l = []
with open(file, 'r') as f:
    for line in f.readlines():
        l.append(line)
        
mol = vermouth.gmx.read_itp(l, ff)

...add modifications...

mol_out = vermouth.Molecule(ff.blocks['name'])

mol_out.meta['moltype'] = 'test'
mol_out.nrexcl = 1

vermouth.gmx.write_molecule_itp(mol_out, outfile='test_out.itp')

```

I would have expected this to write the itp out, but currently I receive:

```
  File /usr/local/lib/python3.8/dist-packages/vermouth/gmx/itp.py:140 in write_molecule_itp
    outfile.write('[ moleculetype ]\n')

AttributeError: 'str' object has no attribute 'write'
```
as the error. 

In short, how can I write an itp file out like this?

**Comment by pckroon** [comment]
Hi, the outfile parameter should be a file opened for writing. So:

```py
with open('test_out.itp', 'w') as outfile:
    vermouth.gmx.write_molecule_itp(mol_out, outfile=outfile)
```

**Reply by csbrasnett** [reply]
Thanks Peter! I can seem to write out a file now, but it only contains the [ moleculetype ] and [ atoms ] directives, without any of the other directives.

I can see that they're all present as interactions in the ff.blocks['name'], but not in the mol_out molecule.

**Reply by fgrunewald** [reply]
@csbrasnett:

to generate a molecule from a block use:
```
mol_out = ff.blocks['name'].to_molecule()
```

I think then it should work. Otherwise the interactions are probably dropped.

**Reply by csbrasnett** [reply]
Thanks Fabian, all sorted now!
